### DISCUSSION TITLE
PDF User's Manual - download here

### DISCUSSION THREAD

**Discussion by dalers** [timschofield/webERP#273] [Development Discussion & Specification]
I simply like having a single PDF open in a viewer to search and browse as the opportunity rises, so I made one including a table of contents (content as of 2024-11-26).

@timschofield if you think the PDF has value for others you can do with it as you wish (e.g. provide it as a download resource on weberp.org, make this post sticky or change the category, etc.).

Others, please comment if you find this PDF of use, comments are the only way to know....

Cheers,
Dale

[webERP Project - Users Manual (2024-11-26).pdf](https://github.com/user-attachments/files/18014639/webERP.Project.-.Users.Manual.2024-11-26.pdf)

**Comment by andrewcouling** [comment]
Coincidentally, I was just looking at the older version of the PDF manual! It's good to have the functional TOC with hyperlinks.

**Reply by dalers** [reply]
Glad you like it. I find I need an indented table of contents to understand the structure of a document, looking at an individual section may help solve an immediate problem, but doesn't help me understand the big picture.

**Comment by pakricard** [comment]
Excellent!

Read it a bit in diagonal and found:

> a user with role "member of parlament" could have a tab of a type "all allowed, including night drinks in a pub", while a user with role "hard worker" could have a tab of a type "transport to workplace" only,

A manual that reflects real life. ;-) Loved it.

**Comment by timschofield** [comment]
Thanks Dale, I have added this to the Documentation page on the web site.

Thanks
Tim

On Wed, 4 Dec 2024 at 22:43, pakricard ***@***.***> wrote:

> Excellent!
>
> Read it a bit in diagonal and found:
>
> a user with role "member of parlament" could have a tab of a type "all
> allowed, including night drinks in a pub", while a user with role "hard
> worker" could have a tab of a type "transport to workplace" only,
>
> A manual that reflects real life. ;-) Loved it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/273#discussioncomment-11466784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LAMS6X4OMCOB6YB3JL2D6ASBAVCNFSM6AAAAABTBDPUQKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBWGY3TQNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by craigbass76** [comment]
Interesting... How'd you make it? I've got a couple repos with PDF creation schemes (bash/pandoc/weasyprint - if anyone's got a way to get a search engine into the pahkahcms one, that'd be great. I gave up and went to a PHP system).

**Comment by dalers** [comment]
Brute force. Manually concatenated HTML files using LibreOffice Writer, manually applied styles and image formatting and created the TOC (fwiw, I was using the PDF manual yesterday and thought the page numbering in at least one of the TOCs was off...).

> I've got a couple repos with PDF creation schemes

Do you think the webERP project could do something similar?

> pahkahcms

Looks interesting. Iiuc, @timschofield used Hugh (https://gohugo.io/) to generate the original version of the new weberp.org website and then hand-edited since. Do you think there would be advantages using pahkahCMS to create/manage weberp.org?

**Reply by craigbass76** [reply]
Like I said - I couldn't figure out how to make a good search engine in Bash. There's a bit of a ruckus going on, but you can see the end result of PahkahCMS here for another few days probably:
https://lbmliftoff.com/oldDocs/documentation01/

**Comment by dalers** [comment]
That's very cool! The download PDF has nice functionality. My best guess after some research is that LBM LIFTOFF uses PahkahCMS to create/publish an online manual (organized by ERP "module") and automatically generates clickable PDFs. Do I understand correctly?

[LBM LIFTOFF](https://lbmliftoff.com) also looks very interesting as ERP software itself. Just curious, what brings you to the webERP community?

**Reply by craigbass76** [reply]
Long story - and I just spilled it all to Tim in a private email... LIFT OFF is no longer a viable option for end users.

The CMS was essentially https://github.com/craigbass76/pandoc-css-weasyprint-template on steroids. I'd figured out how to make nice looking PDFs (to the graphic designer's specs - my cohort was using some Adobe product that kept crapping out, and literally cried when I showed this other way in a team meeting) when I worked at Linux Academy (since sold to ACloudGuru, and since sold to Pluralsite), and took it a few steps farther while working at LIFT OFF. 

But because of the missing search functionality, and my superiors wanting it behind a login, I ended up writing a PHP version. If someone wants to print a doc, it doesn't look as nice. I had to make a whole other page that used a different css stylesheet. I'm working (now that I'm just about unemployed) on getting that PHP version up on GitHub, but haven't gotten there yet. And I'd really like to make the nicer looking PDFs on the fly  with the pandoc/weasyprint method, but hadn't figured out how to do it in PHP on shared web hosting yet. I was also doing data conversions and training, so I got sidetracked a lot.

Something else you might be interested in (once I get it up here on GitHub) is a training matrix. You add a user, and specify what role they're in, and then there's a list of tasks that they need to be trained on. Then you can mark each task, on each employee, as not trained, customer says they're competent, and "we've verified they're competent." Our training area modules mirror the docs modules.

I'm trying to roll out a whole "docs and training," repo with all of it. Shouldn't be too hard to make it work in this ERP. 

So, for example, the company (our customer) is called Acme Inc. They have an employee called Wile E. Coyote, who's in the BirdChaser role. You define what tasks that role has to perform/learn ahead of time (TNT usage, Bat-suit flying, Impromptu tunnel painting, Anvil catching, Cliff diving, etc.), and then you can mark them "*Nope*,  *they say yep*, and *we've confirmed a  yep*,". 

I should have this up in a few days for folks. And I'm hoping WebERP is going to work for my current scenario. I'd love for it work work out, and be able to help with docs.

Apologies to anyone who might take offense - I've kind of hijacked this thread...

**Reply by dalers** [reply]
Thanks for elaborating. The training matrix you described is more generically a "competency" matrix". Roles (generally) and work instructions (the most specific level ) have competencies required for execution and employees are assigned competencies by completing internal training, possessing an industry or academic accreditation or government issued license, etc.

> the company (our customer) is called Acme Inc

My demo company is the "Swift Construction Company" (if you are old enough you may be familar with it). The SCC has 22 staff, which seems to be about the minimum to cover typical engineering, manufacturing, sales, product management, finance and quality. 

> whole "docs and training" repo

How do you control the docs? Fwiw I'm using [SeedDMS](https://www.seeddms.org/) an attractive and functional open source PHP DMS.

**Reply by dalers** [reply]
P.S.

> Apologies to anyone who might take offense - I've kind of hijacked this thread...

No worries, as the OP I don't mind if the discussion wanders....

**Reply by craigbass76** [reply]
There's no version control. And if you want to shuffle the order they're in, you have to rename the markdown files (there were always three numbers preceeding the doc title) and run a script that rebuilt the directory contents. It was kind of a hack, but the best I could crank out in a hurry.

I'm afraid the Swift joke went over my head, unless you're talking the old Tom Swift books with the funny dialog tweaks (_"You're such a bum," Tom said cheekily_ or _"I dropped the toothpaste," Tom said, crestfallen_).

**Reply by dalers** [reply]
> I'm afraid the Swift joke went over my head

Clearly not...  ;-)   Are those your dialog tweaks?

**Reply by craigbass76** [reply]
Heh - not the second one. Swifty came up last week in the Write the Docs watercooler channel (on slack in relation to Taylor Swift though, I brought up Tom Swift and it went downhill from there - it happens when there are that many language geeks in the same place). I did cough up the first one, but there was another tech writer that made up the toothpaste bit.
